File: BE/MoviesAPI/Dtos/Movies/CreateMovieDto.cs
using MoviesAPI.Dtos.MovieTags;
using System.ComponentModel.DataAnnotations;

namespace MoviesAPI.Dtos.Movies
{
    public class CreateMovieDto
    {
        [Required]
        [MaxLength(Constants.Constants.MaxTitleSize)]
        public string Title { get; set; }
        [Required]
        [MaxLength(Constants.Constants.MaxDescriptionSize)]
        public string Description { get; set; }
        [Required]
        public string PosterImageUrl { get; set; }
        [Required]
        public DateTime AddedDate { get; set; }
        [Required]
        public string VidGuardId { get; set; }
        [Required]
        public List<Guid> CategoriesIds { get; set; }
        public List<MovieTagDto>? Tags { get; set; }
    }

}

File: FE/movies-client/src/app/dtos/CreateMovieDto.ts
import { TagDto } from "./TagDto";




export class CreateMovieDto 
{
    public Id!:string;
    public Title!:string;
    public Description!:string;
    public AddedDate!:Date;
    public PosterImageUrl!:string;
    public PosterImageUrlGif!:string;
    public VidGuardId!:string;
    public CategoriesIds!:string[]
    public Tags!:TagDto[];
    public FirebaseId!:string;
}